### PR TITLE
[WIP] add OLM manifests for Operator Hub

### PR DESCRIPTION
**Pull request by zroubalik**
This is a draft of [OLM Manifests](https://github.com/operator-framework/operator-lifecycle-manager/blob/master/doc/design/building-your-csv.md) that will be used to publish Keda on Operator Hub.

**TODO**
- put proper description and info for users
- add other metada

Fixes https://github.com/kedacore/keda/issues/469

### CODE REVIEW

**Comment by zroubalik** [comment]
@ahmelsayed @jeffhollan you might want to check this demo of installing Keda via OLM (the video is to large for embedding as gif)

https://vimeo.com/370203641

**Comment by zroubalik** [comment]
Blocked by OLM issue: https://github.com/operator-framework/operator-lifecycle-manager/issues/1122

**Comment by tomkerkhove** [comment]
> @ahmelsayed @jeffhollan you might want to check this demo of installing Keda via OLM (the video is too large for embedding as gif)
> 
> [vimeo.com/370203641](https://vimeo.com/370203641)

Nice video, looks good!

File: deploy/olm-catalog/keda/0.0.4/keda.v0.0.4.clusterserviceversion.yaml [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+  version: 0.0.4
+  replaces: ''
+  minKubeVersion: '1.11.0'
+  keywords:
+    - keda
+    - scaling
+    - event
+  maintainers:
+    - name: Keda Community
+      email: keda-core@googlegroups.com
```
Comment (tomkerkhove) [inline deploy/olm-catalog/keda/0.0.4/keda.v0.0.4.clusterserviceversion.yaml]: Does this exist? I didn't know that :D

File: deploy/olm-catalog/keda/0.0.4/keda.v0.0.4.clusterserviceversion.yaml [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+  provider:
+    name: Microsoft and Red Hat
```
Comment (tomkerkhove) [inline deploy/olm-catalog/keda/0.0.4/keda.v0.0.4.clusterserviceversion.yaml]: Maybe just list KEDA?

```suggestion
    name: KEDA
```

File: deploy/olm-catalog/keda/0.0.4/keda.v0.0.4.clusterserviceversion.yaml [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+  labels:
+    alm-owner-keda: keda
+    operated-by: keda
+  selector:
+    matchLabels:
+      alm-owner-keda: keda
+      operated-by: keda
+  links:
+    - name: Keda Source Code
```
Comment (tomkerkhove) [inline deploy/olm-catalog/keda/0.0.4/keda.v0.0.4.clusterserviceversion.yaml]: I would add a link to the docs - https://keda.sh

File: deploy/olm-catalog/keda/0.0.4/keda.v0.0.4.clusterserviceversion.yaml (hunk 1 above)
Comment (zroubalik) [reply deploy/olm-catalog/keda/0.0.4/keda.v0.0.4.clusterserviceversion.yaml]: @tomkerkhove I don't know, but @jeffhollan gave me that :)

**Comment by zroubalik** [comment]
I will close this PR, since there are problems on OperatorHub side with installation of our APIServer.
I am creating simple operator that does the installation on OperatorHub for us. Will close this once the repo with the operator is up

**Comment by zroubalik** [comment]
Closing this in favor of https://github.com/kedacore/keda-olm-operator

**Pull request closed without merging**